subsys/bluetooth/host/keys_br.c

```diff
@@ -52,31 +59,66 @@ struct bt_keys_link_key *bt_keys_get_link_key(const bt_addr_t *addr)
 	}
 
 	key = bt_keys_find_link_key(BT_ADDR_ANY);
+#if IS_ENABLED(CONFIG_BT_KEYS_OVERWRITE_OLDEST)
+	if (!key) {
+		int i;
+
+		for (i = 1; i < ARRAY_SIZE(key_pool); i++) {
+			struct bt_keys_link_key *current = &key_pool[i];
+
+			if (current->aging_counter < key->aging_counter) {
+				key = current;
+			}
+		}
+
+		if (key) {
+			bt_keys_link_key_clear(key);
+		}
+	}
+#endif
+
 	if (key) {
 		bt_addr_copy(&key->addr, addr);
+#if IS_ENABLED(CONFIG_BT_KEYS_OVERWRITE_OLDEST)
+		key->aging_counter = ++aging_counter_val;
+		last_keys_updated = key;
+#endif
 		BT_DBG("created %p for %s", key, bt_addr_str(addr));
-
 		return key;
 	}
 
-	BT_DBG("unable to create link key for %s", bt_addr_str(addr));
+	BT_DBG("unable to create keys for %s", bt_addr_str(addr));
 
 	return NULL;
 }
 
 void bt_keys_link_key_clear(struct bt_keys_link_key *link_key)
 {
-	BT_DBG("%s", bt_addr_str(&link_key->addr));
+	if (IS_ENABLED(CONFIG_BT_SETTINGS)) {
+		char key[BT_SETTINGS_KEY_MAX];
+		bt_addr_le_t le_addr;
+
+		le_addr.type = BT_ADDR_LE_PUBLIC;
+		bt_addr_copy(&le_addr.a, &link_key->addr);
+		bt_settings_encode_key(key, sizeof(key), "link_key",
+			&le_addr, NULL);
```

Comment:
Alignment seems wrong here: use tabs and 0-7 spaces to align with the first character after `(` on the first line.